src/w3d/renderer/w3d_util.cpp

```diff
@@ -0,0 +1,73 @@
+/**
+ * @file
+ *
+ * @author Jonathan Wilson
+ *
+ * @brief W3D Utillity class
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "w3d_util.h"
+#include "shader.h"
+#include "vector3.h"
+#include "vector4.h"
+#include "w3d_file.h"
+
+void W3dUtilityClass::Convert_Vector(const IOVector3Struct &v, Vector3 *set)
+{
+    set->X = v.x;
+    set->Y = v.y;
+    set->Z = v.z;
+}
+
+void W3dUtilityClass::Convert_Color(const W3dRGBStruct &rgb, Vector3 *set)
+{
+    set->X = (float)rgb.r / 255;
+    set->Y = (float)rgb.g / 255;
+    set->Z = (float)rgb.b / 255;
+}
+
+void W3dUtilityClass::Convert_Color(const Vector3 &v, W3dRGBStruct *color) {}
+
+void W3dUtilityClass::Convert_Color(const W3dRGBAStruct &rgb, Vector4 *set)
+{
+    set->X = (float)rgb.R / 255;
+    set->Y = (float)rgb.G / 255;
+    set->Z = (float)rgb.B / 255;
+    set->W = (float)rgb.A / 255;
+}
+
+void W3dUtilityClass::Convert_Shader(const W3dShaderStruct &shader, ShaderClass *set)
+{
+    set->Set_Depth_Compare((ShaderClass::DepthCompareType)W3d_Shader_Get_Depth_Compare(&shader));
```

Comment:
Is there no way these casts can be moved up into the Get functions.

Reply:
Nope.